### PR TITLE
feat: add `jina hub list` CLI for displaying executor details

### PR DESCRIPTION
**Pull request by rutujasurve94**
-  - Provides `HubIO` CLI for collecting executor information from `MongoDB` collection
- -  Usage : `jina hub list --type=pod --kind=indexer --name=DummyIndexer --keywords=['sklearn']`

Supports params:

- [x]  name
- [x]  kind
- [x]  type
- [x]  keywords

-  Changes to `HubIO` and `parser` files

- [x] Tested the query in Mongo for filtering `manifest_info` from `build_logs`, it runs successfully.

### CODE REVIEW

File: jina/docker/hubio.py [hunk 1]
```diff
@@ -178,6 +178,36 @@ def login(self):
...
+            keyword_query = {"manifest_info.keywords":  { $all: keywords } }
+            sub_query = f'{keyword_query}, {name_query}'
+
+        _executor_query = '{ $and: [' + f"""{sub_query}""" + ' ] }'
+        with MongoDBHandler(hostname=os.environ['JINA_DB_HOSTNAME'],
+                            username=os.environ['JINA_DB_USERNAME'],
+                            password=os.environ['JINA_DB_PASSWORD'],
+                            database_name=os.environ['JINA_DB_NAME'],
+                            collection_name=os.environ['JINA_DB_COLLECTION']) as db:
+            existing_doc = db.find(query=_executor_query)
```
Comment (deepankarm) [inline jina/docker/hubio.py:206]: Currently `db.find` does `find_one`. Should we limit ourselves to just 1 result?

File: jina/docker/hubio.py [hunk 2]
```diff
@@ -178,6 +178,36 @@ def login(self):
...
+            if existing_doc:
+                executor_detail = existing_doc['manifest_info']
+                self.logger.debug(f'Listing executor info. {executor_detail}')
```
Comment (deepankarm) [inline jina/docker/hubio.py:209]: Should we be showing the complete manifest info present in the database? And how about the info stored about the docker image?

File: jina/docker/hubio.py [hunk 3]
```diff
@@ -178,6 +178,36 @@ def login(self):
...
+            kind_query = {'manifest_info.kind': kind}
+            sub_query = f'{kind_query}'
+        if type:
+            type_query = {'manifest_info.type': type}
+            sub_query = f'{sub_query}, {type_query}'
+        if name:
+            name_query = {'manifest_info.name': name}
+            sub_query = f'{sub_query}, {name_query}'
+        if keywords:
+            keyword_query = {"manifest_info.keywords":  { $all: keywords } }
```
Comment (deepankarm) [inline jina/docker/hubio.py:197]: Shouldn't we do `$any` for keywords, not `$all`?

File: jina/docker/hubio.py [hunk 4]
```diff
@@ -178,6 +178,36 @@ def login(self):
             # use default login
             self._client.login(**get_default_login(), registry=self.args.registry)
 
+    def list(self, name: str = None, kind: str = None, type: str = 'pod', keywords: list = ["numeric", "sklearn"]):
```
Comment (deepankarm) [inline jina/docker/hubio.py:181]: type `str` defaulting to `None`

File: jina/docker/hubio.py [hunk 5]
```diff
@@ -178,6 +178,36 @@ def login(self):
...
+        if not is_db_envs_set():
+            self.logger.error('DB environment variables are not set! aborting list command.')
+            return
+
+        if kind:
+            kind_query = {'manifest_info.kind': kind}
+            sub_query = f'{kind_query}'
+        if type:
+            type_query = {'manifest_info.type': type}
+            sub_query = f'{sub_query}, {type_query}'
```
Comment (deepankarm) [inline jina/docker/hubio.py:192]: If the 1st condition fails, `{sub_query}` wouldn't be defined, will raise an error here.

**Review by hanxiao — CHANGES_REQUESTED** [review]
- missing unit test
- CLI does not work at all

File: jina/docker/hubio.py (hunk 1 above)
Comment (rutujasurve94) [reply jina/docker/hubio.py]: Fitting for multiple results.

File: jina/docker/hubio.py (hunk 2 above)
Comment (rutujasurve94) [reply jina/docker/hubio.py:209]: yes, can be extended over docker info, discussing on channel

File: jina/docker/hubio.py (hunk 3 above)
Comment (rutujasurve94) [reply jina/docker/hubio.py]: done

File: jina/docker/hubio.py (hunk 4 above)
Comment (rutujasurve94) [reply jina/docker/hubio.py:181]: name is default to none.

File: jina/docker/hubio.py (hunk 5 above)
Comment (rutujasurve94) [reply jina/docker/hubio.py:192]: yes, corrected it.

**Comment by rutujasurve94** [comment]
Invoke CLI as:  `jina hub list --remote --type=pod --kind=encoder --keywords=['sklearn', 'numeric']` returns result as :
```
          * Name: jinahub/pod.encoder.randomsparseencoder
          |--   Description: RandomSparseEncoder encodes data from an ndarray in size `B x T` into an ndarray in size `B x D`
          |--   Kind: encoder
          |--   Keywords: ['numeric', 'sklearn']
          |--   Type: pod
          * Name: jinahub/pod.encoder.dummy_mwu_encoder
          |--   Description: a minimum working unit of a containerized encoder, used for tutorial only
          |--   Kind: encoder
          |--   Keywords: ['numeric', 'sklearn']
          |--   Type: pod
```

**Comment by rutujasurve94** [comment]
Closing since `list` is integrated as an AWS Lambda function

**Pull request closed without merging**